src/box/sql/expr.c

```diff
@@ -3176,13 +3176,10 @@ expr_code_int(struct Parse *parse, struct Expr *expr, bool is_neg,
 	const char *sign = is_neg ? "-" : "";
 	if (z[0] == '0' && (z[1] == 'x' || z[1] == 'X')) {
 		errno = 0;
-		if (is_neg) {
+		if (is_neg)
 			value = strtoll(z, NULL, 16);
-		} else {
+		else
 			value = strtoull(z, NULL, 16);
-			if (value > INT64_MAX)
-				goto int_overflow;
-		}
```

Comment:
This `else` block makes no sense now. You should use `strtoll` in both cases otherwise you may get an undetected overflow.

Reply:
I disagree. At the moment, we can work with all INTEGER values supported by Tarantool:
```Lua
tarantool> box.execute([[select 0xFFFFFFFFFFFFFFFF;]])
---
- metadata:
  - name: COLUMN_1
    type: integer
  rows:
  - [18446744073709551615]
...

tarantool> box.execute([[select -0x7FFFFFFFFFFFFFFF;]])
---
- metadata:
  - name: COLUMN_1
    type: integer
  rows:
  - [-9223372036854775807]
...
```

Let's assume that I appy this patch:
```Diff
diff --git a/src/box/sql/expr.c b/src/box/sql/expr.c
index 78a4c4541..d4f757878 100644
--- a/src/box/sql/expr.c
+++ b/src/box/sql/expr.c
@@ -3179,10 +3179,7 @@ expr_code_int(struct Parse *parse, struct Expr *expr, bool is_neg,
        const char *sign = is_neg ? "-" : "";
        if (z[0] == '0' && (z[1] == 'x' || z[1] == 'X')) {
                errno = 0;
-               if (is_neg)
-                       value = strtoll(z, NULL, 16);
-               else
-                       value = strtoull(z, NULL, 16);
+               value = strtoll(z, NULL, 16);
                if (errno != 0) {
                        diag_set(ClientError, ER_HEX_LITERAL_MAX, sign, z,
                                 strlen(z) - 2, 16);
```

Then we will not be able to work with INTEGER values greater than INT64_MAX:
```Lua
tarantool> box.execute([[select 0xFFFFFFFFFFFFFFFF;]])
---
- null
- Hex literal 0xFFFFFFFFFFFFFFFF length 16 exceeds the supported limit (16)
...
```